### DISCUSSION TITLE
How to modify the solver time limit in a CpSolverSolutionCallback?

### DISCUSSION THREAD

**Discussion by haoming-codes** [google/or-tools#4021] [CP-SAT questions]
In python CP-SAT, I'd like to modify the solver time limit in a `CpSolverSolutionCallback.on_solution_callback()`, to eventually achieve an early stopping strategy along the lines of "stopping the search when `current_time - last_solution_time > last_solution_time - start_time`". Is this possible to do? Thanks in advance.

**Comment by lperron** [comment]
You cannot directly.

Have a thread reference the solver with a timer that will call StopSearch().
And have the callback change the deadline of that timer.
